src/drivers/bitbucket_cloud.js

```diff
@@ -199,18 +202,18 @@ class BitBucketCloud {
 
   async paginatedRequest(opts = {}) {
     const { method = 'GET', body } = opts;
-    const result = await this.request(opts);
+    const { next, values } = await this.request(opts);
 
-    if (result.next) {
-      const next = await this.paginatedRequest({
-        fullEndpoint: result.next,
+    if (next) {
+      const nextValues = await this.paginatedRequest({
+        url: next,
         method,
         body
       });
-      result.values.push(...next);
+      values.push(...nextValues);
```

Comment:
why the three dots?

```js
values.push({ ...nextValues });
```

or (if you dont need a deep copy)

values.push(nextValues);

does this even work 🤔 ?

Reply:
Your suggested `values.push(nextValues);` doesn't produce the same effect: we intend to concatenate arrays, not nest them. And `values.push({ ...nextValues });` doesn't behave as expected with arrays either.

Reply:
ah! I thought it was an object.

Reply:
No, it's the `values` array, that contains many objects. We're using tail recursion to append each `next` page values to the end of the results returned by the previous page.